### PR TITLE
In chaos-monkey mode, only keep the mozbrowser pipeline alive.

### LINKED ISSUE #10897: Chaos monkey mode should only protect mozbrowser roots from closure
At the moment `--random-pipeline-closure-probability=P` doesn't kill the root pipeline. It should only do this when pref `dom.mozbrowser.enabled` is true.

### PR DESCRIPTION
**Pull request by asajeffrey**
Fixes #10897

---

This change is [Reviewable](https://reviewable.io/reviews/servo/servo/10898)

### CODE REVIEW

File: components/compositing/constellation.rs [hunk 1]
```diff
@@ -1869,12 +1869,17 @@ impl<LTF: LayoutThreadFactory, STF: ScriptThreadFactory> Constellation<LTF, STF>
...
                 if let Some(pipeline) = self.pipelines.get(pipeline_id) {
-                    // Don't kill the root pipeline
-                    if pipeline.parent_info.is_none() { return; }
-                    // Note that we deliberately do not do any of the tidying up
-                    // associated with closing a pipeline. The constellation should cope!
-                    info!("Randomly closing pipeline {}.", pipeline_id);
-                    pipeline.force_exit();
+                    // Don't kill the mozbrowser pipeline
+                    if prefs::get_pref("dom.mozbrowser.enabled").as_boolean().unwrap_or(false) &&
+                        pipeline.parent_info.is_some()
```
Comment (jdm) [inline components/compositing/constellation.rs]: This looks like it doesn't close child pipelines :<

Comment (asajeffrey) [inline components/compositing/constellation.rs]: This is deliberate, it calls `pipeline.force_exit()` to partly shut down the pipeline, but leaves everything in a deliberately horrible state. Or are you worried that the state isn't horrible enough?

**Comment by asajeffrey** [comment]
An update to #10862

File: components/compositing/constellation.rs (hunk 1 above)
Comment (jdm) [inline components/compositing/constellation.rs]: No, I mean that now only the root pipeline will be forced closed.

Comment (asajeffrey) [inline components/compositing/constellation.rs]: Oops, I flip-flopped on that if test, and ended up flopping where I should have flipped.

**Comment by Manishearth** [comment]
Forgot about this, sorry, r=me

**Pull request merged**